crates/runtime/src/component/mod.rs

```diff
@@ -28,8 +30,36 @@ pub enum Error {
 
 pub mod catalog;
 pub mod dataset;
+pub mod metrics;
 pub mod view;
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ComponentType {
```

Comment:
I added this to use as the prefix of the component metrics, i.e. `<component_type>_<rest of metric name>`.

This could potentially be extended to support a more structured component system - instead of the runtime knowing explicitly all of the component types it supports (i.e. DataConnector, etc), it would just support a `Component` that has a trait interface it uses to register itself.